generator/gpt2/gpt2_generator.py

```diff
@@ -16,6 +15,42 @@
 logging.getLogger("transformers.modeling_utils").setLevel(logging.WARN)
 logging.getLogger("transformers.configuration_utils").setLevel(logging.WARN)
 
+# This will be in the next transformers release, delete then
+def top_k_top_p_filtering(logits, top_k=0, top_p=1.0, filter_value=-float("Inf"), min_tokens_to_keep=1):

```

Comment:
Shouldn't this be inside the model class?

Reply:
Ideally we would import it from transformers.model_utils but it only got added in the latest git version. So it's missing in the latest version - 2.3.0. We would have to install a specific git commit in order to import it.

I mean that's fine to do, just need to change requirements.txt.

Reply:
But it's called as self.top_k_top_p_filtering in the generate method.

Reply:
Whoops, nevermind, I was thinking of something else!